src/common/transport.cc

```diff
@@ -546,28 +578,50 @@ namespace Pistache::Tcp
     {
         int tcp_no_push  = 0;
         socklen_t len    = sizeof(tcp_no_push);
-        int sock_opt_res = getsockopt(GET_ACTUAL_FD(fd), tcp_prot_num_,
-#ifdef __APPLE__
-                                      TCP_NOPUSH,
+        int sock_opt_res = -1;
+
+#ifdef __NetBSD__
+        { // encapsulate
+            int tcp_nodelay = 0;
+            sock_opt_res    = getsockopt(GET_ACTUAL_FD(fd), tcp_prot_num_,
+                                         TCP_NODELAY, &tcp_nodelay, &len);
+            if (sock_opt_res == 0)
+                tcp_no_push = !tcp_nodelay;
+        }
 #else
-                                      TCP_CORK,
+        sock_opt_res = getsockopt(GET_ACTUAL_FD(fd), tcp_prot_num_,
+#if defined __APPLE__ || defined _IS_BSD
+                                  TCP_NOPUSH,
+#else
+                                  TCP_CORK,
 #endif
-                                      &tcp_no_push, &len);
+                                  &tcp_no_push, &len);
+#endif // of ifdef __NetBSD__ ... else
```

Comment:
The prior Pistache code use the Linux-specific MSG_MORE flag with ::send. As per:
https://www.man7.org/linux/man-pages/man2/send.2.html
MSG_MORE is just like the TCP_CORK socket option, except that MSG_MORE is conveniently available to be set, or not set, for every ::send.

Since macOS and BSD don't support MSG_MORE, we simulate it each time we do a ::send, by setting, or resetting, if needed, the TCP_CORK/TCP_NOPUSH socket option.

Next, note that TCP_NOPUSH means the same as TCP_CORK, except that TCP_CORK is the Linux constant, whereas TCP_NOPUSH is for macOS/BSD.

Also, note that if _USE_LIBEVENT_LIKE_APPLE is set in a Linux build, then the more portable TCP_CORK socket option will be used instead of using the more Linux-specific MSG_MORE ::send option even in Linux.

Also, note that when setting a socket flag, we use getsocketopt first, to see the existing flag value, before calling setsocketopt only if the value needs to be changed.

Plus, NetBSD not only does not support MSG_MORE, it doesn't support TCP_CORK/TCP_NOPUSH either. The closest thing it has is TCP_NODELAY.

Now TCP_NODELAY is a little bit like an opposite of TCP_CORK, but it is not an exact opposite.

TCP_CORK/TCP_NOPUSH  causes ::send data to be accumulated, usually to a fixed size limit, before sending on the wire.

TCP_NODELAY turns off Nagle's TCP algorithm. Under Nagle's algorithm, if a small packet is being sent, and prior packets have not been ACKed by the receiver, the sending kernel will accumulate data (not place it on the wire) until the prior sends are ACKed. The thinking is "Well, if the receiver hasn't ACKed what I already sent them, they can't be in a super hurry to receive my new send, I'll hold my new send for now and maybe I can send it as part of a bigger send soon."

So both "TCP_CORK/TCP_NOPUSH being ON" and "TCP_NODELAY being OFF" will tend to have the sender accumulate data before putting it on the wire, albeit TCP_CORK/TCP_NOPUSH is more aggressive about it.

So the net of all this is:
```
bool msg_more = ...
if (linux)
  if (USE_LIBEVENT_LIKE_APPLE)
    Use TCP_CORK = msg_more
  else
    Use Linux-specific MSG_MORE = msg_more
else
  if __NetBSD__
    Use TCP_NODELAY = !msg_more
  else
    Use TCP_NOPUSH = msg_more
```